### PR TITLE
strided slice

### PR DESCRIPTION
**Pull request by justin1121**

### CODE REVIEW

File: tensorflow_encrypted/protocol/pond.py [hunk 1]
```diff
@@ -434,6 +434,29 @@ def transpose(self, x):
 
         return x_t
 
+    # see https://www.tensorflow.org/api_docs/python/tf/strided_slice for documentation on
+    # the arguments
+    def strided_slice(self, x, *args, **kwargs):
+        node_key = ('strided_slice', x)
+        x_t = _nodes.get(node_key, None)
```
Comment (mortendahl) [inline tensorflow_encrypted/protocol/pond.py]: this `_t` naming was for transpose; maybe replace them with `_slice` for clarity?

File: tensorflow_encrypted/protocol/pond.py [hunk 2]
```diff
@@ -1572,10 +1595,72 @@ def _transpose_masked(prot, x_masked):
...
+
+def _strided_slice_masked(prot, x_masked, args, kwargs):
+    assert isinstance(x_masked, PondMaskedTensor)
+
+    a, a0, a1, alpha_on_0, alpha_on_1 = x_masked.unwrapped
+
+    with tf.name_scope('strided_slice'):
+
+        with tf.device(prot.crypto_producer.device_name):
+            a_t = a.strided_slice()
```
Comment (mortendahl) [inline tensorflow_encrypted/protocol/pond.py]: you're not passing `args` and `kwargs` in neither of the three applications here

File: tensorflow_encrypted/protocol/pond.py [hunk 3]
```diff
@@ -434,6 +434,29 @@ def transpose(self, x):
...
+    # see https://www.tensorflow.org/api_docs/python/tf/strided_slice for documentation on
+    # the arguments
+    def strided_slice(self, x, *args, **kwargs):
+        node_key = ('strided_slice', x)
+        x_t = _nodes.get(node_key, None)
+
+        if x_t is not None:
+            return x_t
+
+        if isinstance(x, PondPublicTensor):
```
Comment (bendecoste) [inline tensorflow_encrypted/protocol/pond.py]: should we use the dispatch pattern from `pond`? maybe not enough types to be worth it

File: tensorflow_encrypted/protocol/pond.py (hunk 2 above)
Comment (justin1121) [reply tensorflow_encrypted/protocol/pond.py]: Code catch 😄

File: tensorflow_encrypted/protocol/pond.py (hunk 3 above)
Comment (justin1121) [reply tensorflow_encrypted/protocol/pond.py]: Other unary ops do it like this.

**Pull request merged**